polaris-react/src/components/Navigation/Navigation.scss

```diff
@@ -290,12 +316,25 @@ $disabled-fade: 0.6;
   display: flex;
   height: nav(mobile-height);
 
+  #{$se23} & {
+    align-items: center;
+    height: calc(nav(mobile-height) - var(--p-space-1));
```

Comment:
Maybe it could be a new variable, but I thought it would be nicer to create it when we are ready to remove the fallback code.